src/figuresClasses.ts

```diff
@@ -1,19 +1,72 @@
 export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea: () => number;
+}
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
 
-}
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a:number,
+    public b:number,
+    public c:number,
+  ) {
+    if (a >= b + c || b >= a + c || c >= a + b || a <= 0 || b <= 0 || c <= 0) {
```

Comment:
may be better to rewrite each condition on a new line

Reply:
and create a separate function for checking is the side's size is below zero. And use this function in other classes.